cpp/src/gandiva/precompiled/extended_math_ops.cc

```diff
@@ -212,6 +230,32 @@ ENUMERIC_TYPES_UNARY(RADIANS, float64)
   }
 ENUMERIC_TYPES_UNARY(DEGREES, float64)
 
+// Abs
+#define ABS(IN_TYPE)                                                      \
+  FORCE_INLINE                                                            \
+  gdv_##IN_TYPE abs_##IN_TYPE(gdv_##IN_TYPE in) {                         \
+    return static_cast<gdv_##IN_TYPE>(abs(static_cast<long double>(in))); \
```

Comment:
I am curious whether all platforms can convert 64-bit integer to `long double` without loss of digits. (i.e. size(long double) > size(int64)).

Reply:
@kiszk I checked that by the `IEE-754` convention, the numbers in double precision [should use at least 8 bytes](https://en.wikipedia.org/wiki/IEEE_754) to construct their representation, so I think that there would not have any digit loss in conversion.